payments/models.py

```diff
@@ -135,6 +135,71 @@ def get_success_url(self) -> str:
     def get_process_url(self) -> str:
         return reverse('process_payment', kwargs={'token': self.token})
 
+    def get_payment_url(self):
+        """
+        Get the url the view that handles the payment (payment_details() in documentation)
+        For now used only by PayU provider to redirect users back to CVV2 form
+        """
+        raise NotImplementedError()
+
+    def get_user(self):
```

Comment:
What's this supposed to return?

Reply:
This is simple helper method used by the other `get_user_*` methods, so it is the only method, that needs to be overridden in case of the Django `django.contrib.auth.models.User` model.

Reply:
You're assuming that all applications have a `User` for each payment made.

I'm not a fan of imposing such a big restriction, especially without the need for one. You probably want a `user` in your own `Payment` instance, not in the abstract one in this package.